### PR TITLE
Replace reinterpret_cast with memcpy in PickingManager.h

### PR DESCRIPTION
**Pull request by ronaldfw**
Also, add some more static_asserts to ensure that the sizes match.

### CODE REVIEW

File: OrbitGl/PickingManager.h [hunk 1]
```diff
@@ -56,11 +56,19 @@ struct PickingID {
     return id;
   }
   static Color GetColor(Type a_Type, uint32_t a_ID, BatcherId batcher_id = TIME_GRAPH) {
+    static_assert(sizeof(PickingID) == sizeof(Color),
+      "PickingId and Color must have the same size");
     PickingID id = Get(a_Type, a_ID, batcher_id);
-    return *(reinterpret_cast<Color*>(&id));
+    Color color;
+    memcpy(&color, &id, sizeof(PickingID));
```
Comment (beckerhe) [inline OrbitGl/PickingManager.h:63]: I think we should use `std::memcpy` and include `<cstring>`.

File: OrbitGl/PickingManager.h [hunk 2]
```diff
@@ -56,11 +56,19 @@ struct PickingID {
     return id;
   }
   static Color GetColor(Type a_Type, uint32_t a_ID, BatcherId batcher_id = TIME_GRAPH) {
+    static_assert(sizeof(PickingID) == sizeof(Color),
+      "PickingId and Color must have the same size");
```
Comment (beckerhe) [inline OrbitGl/PickingManager.h:60]: Nit: I like having that static_assert. Ideally we also check whether the types are `std::is_trivially_copyable`.

File: OrbitGl/PickingManager.h (hunk 1 above)
Comment (ronaldfw) [reply OrbitGl/PickingManager.h]: Done

File: OrbitGl/PickingManager.h (hunk 2 above)
Comment (ronaldfw) [reply OrbitGl/PickingManager.h:61]: Good idea, I added this and found out that Color is not trivially copyable. To be on the safe side, I copy the values in an array first and then call the actual constructor of Color. 

Added static asserts where appropriate, I think (let me know if I missed something).

Comment (beckerhe) [reply OrbitGl/PickingManager.h:61]: Ahh good Point. I remember having that problem before. If I remember correctly Color is a gteVector4, which is not trivially copyable. At some point we should replace gteVector4 by something more appropriate, but for now this change looks good! Thanks.

**Pull request merged**